index.js

```diff
@@ -23,13 +23,34 @@ module.exports.pitch = function (request) {
 		"// Hot Module Replacement",
 		"if(module.hot) {",
 		"	// When the styles change, update the <style> tags",
-		"	if(!content.locals) {",
-		"		module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
-		"			var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
-		"			if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
+		"	module.hot.accept(" + loaderUtils.stringifyRequest(this, "!!" + request) + ", function() {",
+		"		var newContent = require(" + loaderUtils.stringifyRequest(this, "!!" + request) + ");",
+		"		if(typeof newContent === 'string') newContent = [[module.id, newContent, '']];",
+		"		var localsSame = true;",
+		"		if(!content.locals && newContent.locals) {",
+		"			localsSame = false;",
+		"		} else if(content.locals && !newContent.locals) {",
+		"			localsSame = false;",
+		"		} else if(content.locals && newContent.locals) {",
+		"			var oldKeys = Object.keys(content.locals);",
+		"			var newKeys = Object.keys(newContent.locals);",
+		"			if(oldKeys.length !== newKeys.length) {",
+		"				localsSame = false;",
+		"			} else {",
+		"				localsSame = oldKeys.every(function(oldKey) {",
+		"					var oldValue = content.locals[oldKey];",
+		"					var newValue = newContent.locals[oldKey];",
+		"					return oldValue === newValue;",
+		"				});",
+		"			}",
+		"		}",
+		"		if(localsSame) {",
 		"			update(newContent);",
-		"		});",
-		"	}",
+		"		} else {",
+		"			// This error is caught and not shown and causes a full reload.",
+		"			throw new Error('Aborting CSS HMR due to changed css-modules locals.');",
```

Comment:
I looked through https://webpack.js.org/api/hot-module-replacement/ and tried to read the HMR source code in the webpack repo, and could not find any other way of aborting an `accept` other than throwing an error. It seems to work flawlessly, though!

Reply:
I will take a look into this, I think it's fine